### PR TITLE
add integration tests to CI

### PR DESCRIPTION
**Pull request by demisx**
# Goal
The goal of this PR is to enable integration tests in CI.

Part of #852

### CODE REVIEW

File: .github/workflows/verify-pr-commit.yml [hunk 1]
```diff
@@ -38,9 +39,14 @@ jobs:
...
               - '**/Cargo.lock'
               - 'js/api-augment/**'
+              - 'integration-tests/**/*.test.ts'
             cargo-lock:
               - '**/Cargo.toml'
               - '**/Cargo.lock'
+            run-integration-tests:
+              - '**/*.rs'
+              - '**/Cargo.toml'
+              - 'integration-tests/**/*.test.ts'
```
[the remark this answers is not in the document]
Comment (wilwade) [reply .github/workflows/verify-pr-commit.yml:49]: In theory a Cargo change could end up with a broken integration test, but the situations where unit tests don't catch it is extremely small. So I think this is correct, although I would just go ahead and test on any ts or json file in integration tests: (not 100% sure this glob syntax is correct)
```suggestion
              - 'integration-tests/**/*.[ts,json]'
```

File: .github/workflows/verify-pr-commit.yml [hunk 2]
```diff
@@ -69,8 +75,10 @@ jobs:
         git_branch: ["${{github.head_ref}}", main]
         include:
           - git_branch: ${{github.head_ref}}
+            spec: frequency-rococo-local
             branch_name: PR
           - git_branch: main
+            spec: frequency-rococo-local
```
[the remark this answers is not in the document]
Comment (wilwade) [reply .github/workflows/verify-pr-commit.yml:81]: Hmmm... That means our PR build misses the mainnet runtime code. That said we do `verify-rust-packages-and-deps` with all features and of course our other builds are still building it. So I think it is ok.

File: integration-tests/msa/createMsa.test.ts [hunk 3]
```diff
@@ -1,69 +1,82 @@
...
+    const retireMsaEvents = await signAndSend(() => api.tx.msa.retireMsa(), keys);
 
-        assert.notEqual(retireMsaEvents["msa.PublicKeyDeleted"], undefined, 'should have deleted public key (retired)');
-        assert.notEqual(retireMsaEvents["msa.MsaRetired"], undefined, 'should have retired msa');
-    });
-})
+    assert.notEqual(retireMsaEvents["msa.PublicKeyDeleted"], undefined, "should have deleted public key (retired)");
+    assert.notEqual(retireMsaEvents["msa.MsaRetired"], undefined, "should have retired msa");
+  });
+});
```
[the remark this answers is not in the document]
Comment (wilwade) [reply integration-tests/msa/createMsa.test.ts:82]: It has a `npm run format`, but we might need CI to enforce it by testing `npm run lint`.

File: .github/workflows/verify-pr-commit.yml [hunk 4]
```diff
@@ -585,3 +593,68 @@ jobs:
...
+
+  run-integration-tests:
+    if: needs.changes.outputs.run-integration-tests == 'true'
+    needs: build-binaries
+    name: Run Integration Tests
+    runs-on: ubuntu-20.04
+    env:
+      REF_TAG: latest # tag to obtain reference binary to test against
+      NETWORK: mainnet
+      GITHUB_PR_LABEL: metadata-mismatch
```
Comment (wilwade) [inline .github/workflows/verify-pr-commit.yml:605]: I think these are extra:
```suggestion
```

**Review by wilwade — CHANGES_REQUESTED** [review]
> I currently build the main binary against the latest PR commit, then start local node in instant seal mode and run integration tests against it. Is this sufficient to meet this AC?

Yes.



>> (May want to parse out to a different issue) Should still work if there are api-augment changes without needing an npm release of api-augment. (aka should build and use "local" api-augment)

> I am not sure what needs to be done in the run-integration-tests in order to support this.

I'll see if I can parse what it needs a bit more. Doesn't need to be in this PR however.

**Review by saraswatpuneet — APPROVED** [review]
Looks good

**Review by wilwade — APPROVED** [review]

**Pull request merged**